### DISCUSSION TITLE
Expected running time for FPGrowth

### DISCUSSION THREAD

**Discussion by sAslesh** [rasbt/mlxtend#858] [Q&A]
Hi, 

I am using the FPGrowth algorithm for my research project. The dataset that I am using (webdocs dataset) can be found here - http://fimi.uantwerpen.be/data/ 
The dataset has 1,692,082 transactions and 5,267,567 items. I tried to run the fpgrowth API for it and the total time taken was around 51 minutes. **Just wanted to check if the running time is along the expected lines** 

I am attaching the code as well - [fp.txt](https://github.com/rasbt/mlxtend/files/7602764/fp.txt)

Thanks in advance.

**Comment by rasbt** [comment]
Thanks for mentioning. Given such a large dataset, the runtime might be expected. You could try `fpmax`, which should be faster if needed.
